### PR TITLE
Docs: clarify where counter variables start

### PR DESCRIPTION
**Pull request by ghiculescu**
The way the guide is written now, it's not clear if the first counter value will be `0` or `1`, so I made it explicit that it's `0`.

### CODE REVIEW

File: guides/source/layouts_and_rendering.md [hunk 1]
```diff
@@ -1328,7 +1328,7 @@ You can also pass in arbitrary local variables to any partial you are rendering
...
 
-TIP: Rails also makes a counter variable available within a partial called by the collection, named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter` which indexes the number of times it has been rendered within the enclosing view. Note that it also applies for when the partial name was changed by using the `as:` option. For example, the counter variable for the code above would be `item_counter`.
+TIP: Rails also makes a counter variable available within a partial called by the collection, named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter` which indexes the number of times it has been rendered within the enclosing view, starting with a value of `0` on the first render. Note that it also applies for when the partial name was changed by using the `as:` option. For example, the counter variable for the code above would be `item_counter`.
```
Comment (p8) [inline guides/source/layouts_and_rendering.md:1331]: Hey @ghiculescu 👋 .
This whole TIP is a bit difficult to understand/visualize.
What do you think about adding an example (and not render it as a TIP)?

**Comment by ghiculescu** [comment]
image
this is unrelated

File: guides/source/layouts_and_rendering.md [hunk 2]
```diff
@@ -1328,12 +1328,26 @@ You can also pass in arbitrary local variables to any partial you are rendering
...
 
-TIP: Rails also makes a counter variable available within a partial called by the collection, named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter` which indexes the number of times it has been rendered within the enclosing view. Note that it also applies for when the partial name was changed by using the `as:` option. For example, the counter variable for the code above would be `item_counter`.
+#### Counter Variables
 
-You can also specify a second partial to be rendered between instances of the main partial by using the `:spacer_template` option:
+Rails also makes a counter variable available within a partial called by the collection, named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter` which indexes the number of times it has been rendered within the enclosing view, starting with a value of `0` on the first render.
```
Comment (p8) [inline guides/source/layouts_and_rendering.md:1333]: Some of these sentences are pretty long.
What do you think about cutting them up a bit?
Maybe something like:
```suggestion
Rails also makes a counter variable available within a partial called by the collection. The variable is named after the title of the partial followed by `_counter`. For example, when rendering a collection `@products` the partial `_product.html.erb` can access the variable `product_counter`. The variable indexes the number of times the partial has been rendered within the enclosing view, starting with a value of `0` on the first render.
```

**Comment by ghiculescu** [comment]
ooops, i've done something wrong to this branch.... let me start fresh.

**Pull request closed without merging**